pandas/core/reshape/reshape.py

```diff
@@ -1094,6 +1097,165 @@ def get_empty_frame(data) -> DataFrame:
+    ...                    "col2_c": [0, 0, 0]})
+
+    >>> pd.from_dummies(df, sep="_", implied_category={"col1": "d", "col2": "e"})
+        col1    col2
+    0    a       b
+    1    b       a
+    2    d       e
+    """
+    from pandas.core.reshape.concat import concat
+
+    if data.isna().any().any():
+        raise ValueError(
+            f"Dummy DataFrame contains NA value in column: "
+            f"'{data.columns[data.isna().any().argmax()]}'"
+        )
+
+    # index data with a list of all columns that are dummies
+    try:
+        data_to_decode = data.astype("boolean")
+    except TypeError:
+        raise TypeError("Passed DataFrame contains non-dummy data")
+
+    # collect prefixes and get lists to slice data for each prefix
+    if sep is None:
+        variables_slice = {"": list(data.columns)}
+    elif isinstance(sep, str):
+        variables_slice = {}
+        for col in data_to_decode.columns:
+            prefix = col.split(sep)[0]
+            if len(prefix) == len(col):
+                raise ValueError(f"Separator not specified for column: {col}")
+            if prefix not in variables_slice:
+                variables_slice[prefix] = [col]
+            else:
+                variables_slice[prefix].append(col)
+    else:
+        raise TypeError(
+            f"Expected 'sep' to be of type 'str' or 'None'; "
+            f"Received 'sep' of type: {type(sep).__name__}"
+        )
+
+    # validate number of implied_category
+    def check_len(item, name) -> None:
+        if not len(item) == len(variables_slice):
+            len_msg = (
+                f"Length of '{name}' ({len(item)}) did not match the "
+                "length of the columns being encoded "
+                f"({len(variables_slice)})"
+            )
+            raise ValueError(len_msg)
+
+    if implied_category:
+        if isinstance(implied_category, dict):
+            check_len(implied_category, "implied_category")
+        elif isinstance(implied_category, Hashable):
+            implied_category = dict(
+                zip(variables_slice, [implied_category] * len(variables_slice))
+            )
+        else:
+            raise TypeError(
+                f"Expected 'implied_category' to be of type "
+                f"'None', 'Hashable', or 'dict'; "
+                f"Received 'implied_category' of type: "
+                f"{type(implied_category).__name__}"
+            )
+
+    cat_data = {}
+    for prefix, prefix_slice in variables_slice.items():
+        if sep is None:
+            cats = prefix_slice.copy()
+        else:
+            cats = [
+                col[len(prefix + sep) :] for col in prefix_slice if isinstance(col, str)
+            ]
+        assigned = data_to_decode[prefix_slice].sum(axis=1)
+        if any(assigned > 1):
+            raise ValueError(
+                f"Dummy DataFrame contains multi-assignment(s); "
+                f"First instance in row: {assigned.argmax()}"
```

Comment:
Same here. argmax or idxmax?

Reply:
Thanks for the reminder, `argmax` was just used out of habit